internal/contentcache/contentcache.go

```diff
@@ -50,3 +50,38 @@ func New(tempDir string, mtimeClock timeutil.Clock) *ContentCache {
 func (c *ContentCache) NewTempFile(rc io.ReadCloser) (gcsx.TempFile, error) {
 	return gcsx.NewTempFile(rc, c.tempDir, c.mtimeClock)
 }
+
+// Function to add or update existing cache file
```

Comment:
The cache is going to be used concurrently. Any idea how to ensure thread safety? If the caller is required to ensure it's used in a thread safe way, we should note that in the comment.

Reply:
Will add a comment, this in memory map is not concurrent safe.